base/client.jl

```diff
@@ -210,113 +208,122 @@ function init_bind_addr(args::Vector{UTF8String})
-    i = 1
-    while i <= length(args)
-        if args[i]=="-q" || args[i]=="--quiet"
-            quiet = true
-        elseif args[i]=="--worker"
-            start_worker()
-            # doesn't return
-        elseif args[i]=="--bind-to"
-            i+=1 # has already been processed
-        elseif args[i]=="-e" || args[i]=="--eval"
-            i == length(args) && error("-e,--eval  no <expr> provided")
-            repl = false
-            i+=1
-            splice!(ARGS, 1:length(ARGS), args[i+1:end])
-            eval(Main,parse_input_line(args[i]))
-            break
-        elseif args[i]=="-E" || args[i]=="--print"
-            i == length(args) && error("-E,--print  no <expr> provided")
-            repl = false
-            i+=1
-            splice!(ARGS, 1:length(ARGS), args[i+1:end])
-            show(eval(Main,parse_input_line(args[i])))
-            println()
-            break
-        elseif args[i]=="-P" || args[i]=="--post-boot"
-            i == length(args) && error("-P,--post-boot  no <expr> provided")
-            i+=1
-            eval(Main,parse_input_line(args[i]))
-        elseif args[i]=="-L" || args[i]=="--load"
-            i == length(args) && error("-L, --load  no <file> provided")
-            i+=1
-            require(args[i])
-        elseif args[i]=="-p"
-            i == length(args) && error("-p  <n> processes not provided")
-            i+=1
-            if i > length(args) || !isdigit(args[i][1])
-                np = Sys.CPU_CORES
-                i -= 1
-            else
-                np = int(args[i])
-                np < 1 && error("-p  <n> must be ≥ 1")
+# NOTE: This set of required arguments need to be kept in sync with the required arguments defined in ui/repl.c
+let reqarg = Set(UTF8String["--home",          "-H",
+                            "--eval",          "-e",
+                            "--print",         "-E",
+                            "--post-boot",     "-P",
+                            "--load",          "-L",
+                            "--sysimage",      "-J",
+                            "--cpu-target",    "-C",
+                            "--procs",         "-p",
+                            "--machinefile",
+                            "--color",
+                            "--history-file",
+                            "--startup-file",
+                            "--compile",
+                            "--check-bounds",
+                            "--int-literals",
+                            "--dump-bitcode",
+                            "--depwarn",
+                            "--inline",
+                            "--build",        "-b",
+                            "--bind-to"])
+    global process_options
+    function process_options(opts::JLOptions, args::Vector{UTF8String})
+        if !isempty(args) && (arg = first(args); arg[1] == '-' && in(arg, reqarg))
+            println(STDERR, "julia: option `$arg` is missing an argument")
```

Comment:
couldn't we do this by checking the return value of getopt_long? http://man7.org/linux/man-pages/man3/getopt.3.html#RETURN_VALUE